packages/app-desktop/gui/WindowCommandsAndDialogs/commands/importFrom.ts

```diff
@@ -0,0 +1,173 @@
+		}
+	}
+
+	const importModules = InteropService.instance().modules().filter(module => module.type === ModuleType.Importer) as ImportModule[];
+	return await control.showPrompt({
+		label: _('Select the type of file to be imported:'),
+		value: '',
+		suggestions: importModules.map(module => {
+			const label = module.fullLabel();
+			return {
+				key: `${module.type}--${label}`,
+				value: module,
+				label: module.fullLabel(),
+			};
+		}),
+	});
+};
+
+const promptForSourcePath = async (module: ImportModule, sourceType: FileSystemItem|undefined) => {
+	if (!sourceType) {
+		if (!module.sources.includes(FileSystemItem.Directory)) {
+			sourceType = FileSystemItem.File;
+		}
+		if (!module.sources.includes(FileSystemItem.File)) {
+			sourceType = FileSystemItem.Directory;
+		}
+	}
+	if (sourceType === FileSystemItem.File) {
+		return await bridge().showOpenDialog({
+			filters: [{ name: module.description, extensions: module.fileExtensions }],
+		});
+	} else if (sourceType === FileSystemItem.Directory) {
+		return await bridge().showOpenDialog({
+			properties: ['openDirectory', 'createDirectory'],
+		});
+	} else {
+		return await bridge().showOpenDialog({
+			properties: ['openDirectory', 'openFile'],
+		});
+	}
+};
+
+export const runtime = (control: WindowControl): CommandRuntime => {
+	return {
+		// Since this can be run from "go to anything", partialOptions needs to support being null or empty.
+		execute: async (context: CommandContext, options: ImportCommandOptions|undefined) => {
+			const importModule = await findImportModule(options, control);
+			if (!importModule) return null; // E.g. if cancelled
+
+			let sourcePath = options?.sourcePath ?? await promptForSourcePath(importModule, options?.sourceType);
+			if (Array.isArray(sourcePath)) {
+				sourcePath = sourcePath[0];
+			}
+			// Handle the case where the directory picker action was cancelled
+			if (!sourcePath) return null;
+
+			if (!options) {
+				const isDirectory = await shim.fsDriver().isDirectory(sourcePath);
+				const importsMultipleNotes = importModule.isNoteArchive || isDirectory;
+
+				const destinationFolderId = await showFolderPicker(control, {
+					label: _('Import to notebook:'),
+					// Allow selecting no parent folder only if a new folder will
+					// be created:
+					allowSelectNone: importsMultipleNotes,
+					showFolder: (_folder)=>true,
+				});
+				const importFormat = importModule.format;
+				const outputFormat = importModule.outputFormat;
+				options = {
+					sourcePath,
+					destinationFolderId,
+					importFormat,
+					outputFormat,
+					sourceType: isDirectory ? FileSystemItem.Directory : FileSystemItem.File,
+				};
+			}
+
+			const modalMessage = _('Importing from "%s" as "%s" format. Please wait...', sourcePath, options.importFormat);
+			void CommandService.instance().execute('showModalMessage', modalMessage);
+
+			// eslint-disable-next-line @typescript-eslint/no-explicit-any -- Old code before rule was applied
```

Comment:
This, and much of the other code in `importFrom.ts`, was moved from `MenuBar.tsx`.